domain/service/impl/invitation_service.go

```diff
@@ -10,47 +10,71 @@ import (
 
 type InvitationService struct {
 	invitationStorage IInvitationStorage
-	userStorage       IUserStorage
+	groupStorage      IGroupStorage
 }
 
-func NewInvitationService(invitationStorage *IInvitationStorage, userStorage *IUserStorage) IInvitationService {
-	return &InvitationService{invitationStorage: *invitationStorage, userStorage: *userStorage}
+func NewInvitationService(invitationStorage *IInvitationStorage, groupStorage *IGroupStorage) IInvitationService {
+	return &InvitationService{invitationStorage: *invitationStorage, groupStorage: *groupStorage}
 }
 
-func (i InvitationService) CreateGroupInvitation(request GroupInputDTO, groupID UUID) error {
+func (i InvitationService) CreateGroupInvitation(request GroupInvitationDTO) error {
 	// get invites from request
 	invites := request.Invites
-	// TODO: change, wrong implementation, look up how to store association in gorm
 
-	// handle group invitations
+	// handle group invitations for all invitees
 	for _, invitee := range invites {
-		groupInvitation := model.CreateGroupInvitation(groupID, invitee)
+		groupInvitation := model.CreateGroupInvitation(request.GroupID, invitee)
 		err := i.invitationStorage.AddGroupInvitation(groupInvitation)
 		if err != nil {
 			return err
 		}
 
 	}
 
-	/*
-		// add group invitation to all users
-		for _, userID := range invites {
-			err = i.IUserStorage.AddGroupInvitation(groupInvitation, userID)
-			// TODO: error handling, should return to which users the invitation could not be added
-			core.LogError(err)
-		}
+	return nil
+}
 
-	*/
+func (i InvitationService) GetGroupInvitationByID(id UUID) (GroupInvitationOutputDTO, error) {
+	group, err := i.invitationStorage.GetGroupInvitationByID(id)
+	if err != nil {
+		return GroupInvitationOutputDTO{}, err
+	}
+	return ToGroupInvitationDTO(group), nil
+}
 
-	return nil
+func (i InvitationService) GetGroupInvitationsFromUser(id UUID) ([]GroupInvitationOutputDTO, error) {
+	groupInvitations, err := i.invitationStorage.GetGroupInvitationsByUserID(id)
+	if err != nil {
+		return nil, err
+	}
+	// convert group invitations to data transfer objects
+	var result []GroupInvitationOutputDTO
+	for _, groupInvitation := range groupInvitations {
+		result = append(result, ToGroupInvitationDTO(groupInvitation))
+	}
+	return result, nil
 }
 
-func (i InvitationService) AcceptGroupInvitation(invitation UUID, userID UUID) error {
-	//TODO implement me
-	panic("implement me")
+// TODO: Handle consistency. AddMemberToGroup and DeleteGroupInvitation should be performed both or none.
+func (i InvitationService) AcceptGroupInvitation(invitationID UUID, userID UUID) error {
+	// TODO: is this really necessary?
```

Comment:
This seems fine for me. What is your concern?

Reply:
I mean the "TODO: is this really necessary?" here :)

Reply:
If we would add the groupID to the GroupInvitationInputDTO we could save one sql query. But otherwise, we have to check whether the invitationID belongs to a correct invitation so such a query is necessary. I think we will look into this in the future so for now i would like to leave the TODO.

Reply:
Got it. I think we might need to keep the one additional query anyway, but we can circle back to it someday.